### PR TITLE
fix metaRPCClientFactory id handling

### PR DESCRIPTION
**Pull request by shanejonas**
Explanation:  
these changes affect `null` and `undefined` and `0` handling for `metaRPC` to reflect the [JSON-RPC Spec](https://www.jsonrpc.org/specification)

Rel: https://github.com/MetaMask/json-rpc-middleware-stream/pull/18

### CODE REVIEW

File: app/scripts/lib/metaRPCClientFactory.js [hunk 1]
```diff
@@ -17,34 +18,46 @@ class MetaRPCClient {
...
   }
 
+  onUncaughtError(handler) {
+    this.uncaughtErrorChannel.addListener('error', (error) => {
+      handler(error);
+    });
+  }
+
   close() {
     this.notificationChannel.removeAllListeners();
```
Comment (shanejonas) [inline app/scripts/lib/metaRPCClientFactory.js:28]:
```suggestion
    this.notificationChannel.removeAllListeners();
    this.uncaughtErrorChannel.removeAllListeners();
```

Comment (rekmarks) [reply app/scripts/lib/metaRPCClientFactory.js:28]: I endorse this change, lol

File: app/scripts/lib/metaRPCClientFactory.js [hunk 2]
```diff
@@ -17,34 +18,46 @@ class MetaRPCClient {
     });
   }
 
+  onUncaughtError(handler) {
+    this.uncaughtErrorChannel.addListener('error', (error) => {
+      handler(error);
+    });
+  }
```
Comment (rekmarks) [inline app/scripts/lib/metaRPCClientFactory.js:25]: We should use this new method, probably by adding a listener next to [this](https://github.com/MetaMask/metamask-extension/blob/09d81ac/ui/index.js/#L131-L141) in `ui/index.js`.

That should be done in a follow-up, though.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**